share/eds/store.go

```diff
@@ -150,20 +150,21 @@ func (s *Store) gc(ctx context.Context) {
 // The square is verified on the Exchange level, and Put only stores the square, trusting it.
 // The resulting file stores all the shares and NMT Merkle Proofs of the EDS.
 // Additionally, the file gets indexed s.t. store.Blockstore can access them.
-func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.ExtendedDataSquare) error {
-	ctx, span := tracer.Start(ctx, "store/put", trace.WithAttributes(attribute.String("root", root.String())))
-	defer span.End()
+func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.ExtendedDataSquare) (err error) {
+	ctx, span := tracer.Start(ctx, "store/put", trace.WithAttributes(
+		attribute.String("root", root.String()),
+		attribute.Int("width", int(square.Width())),
+	))
+	defer utils.SetStatusAndEnd(span, err)
```

Comment:
I really love the idea here, and my eyes truly opened when I saw this. 
However, this will always be nil because it passes the value at the moment of the defer call.

Here is a snippet to play around https://go.dev/play/p/i2yVr-bkIjy. 

The fix is simple though https://go.dev/play/p/jSA8Kora7xS

Reply:
Ah yes makes sense! Thanks for catching this